src/workerd/api/streams/standard.c++

```diff
@@ -1660,10 +1660,14 @@ ReadableStreamJsController::ReadableStreamJsController(
     : state(consumer.clone(js, this)) {}
 
 ReadableStreamJsController::ReadableStreamJsController(kj::Own<ValueReadable> consumer)
-    : state(kj::mv(consumer)) {}
+    : state(kj::mv(consumer)) {
+  state.get<kj::Own<ValueReadable>>()->setOwner(this);
```

Comment:
Root cause here is that the `ValueReadable` and `ByteReadable` were retaining the pointer to the previous owner. `ReadableStreamJsSource` was handling this case properly but missed it here.